### PR TITLE
Add color And \\ulnone 

### PR DESCRIPTION
**Pull request by zyk-mjzs**
Add color property
Increase judgment on \\ ulnone for cancel underline

Add parse_color_table and parse_underline on parser.rs

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -322,6 +334,46 @@ impl<'a> Parser<'a> {
...
+        let mut table = HashMap::new();
+        let mut current_key = 1;
+        let mut current_color = Color::default();
+        for token in color_table_tokens.iter() {
+            match token {
+                Token::ControlSymbol((control_word, property)) => match control_word {
+                    ControlWord::Unknown(name) => {
+                        if let Property::Value(key) = property {
+                            match *name {
+                                "\\red" => {
```
Comment (d0rianb) [inline src/parser.rs]: `\red` `\green` and `\blue` should be Token

File: src/document.rs [hunk 2]
```diff
@@ -1,6 +1,17 @@
 use crate::header::RtfHeader;
 use crate::parser::StyleBlock;
 
+#[cfg(feature="serde_support")]
+use serde::{Deserialize, Serialize};
+
+#[cfg(feature="serde_support")]
```
Comment (d0rianb) [inline src/document.rs]: Using `#[cfg_attr(feature = "serde_support", derive(Deserialize, Serialize))]` would prevent the code duplication

**Review by d0rianb — COMMENTED** [review]
Thanks for your commit, the code is very clean.
However, I made a few comments on some details.

File: src/document.rs (hunk 2 above)
Comment (zyk-mjzs) [reply src/document.rs]: thx, I got it

**Comment by d0rianb** [comment]
Wonderfull, thanks for all the efforts you put in this PR.

**Pull request merged**